src/pudl/extract/ferc2.py

```diff
@@ -40,6 +41,18 @@ def finalize_schema(self, meta: sa.MetaData) -> sa.MetaData:
             meta, pk_table="f2_s0_respondent_id", column="respondent_id"
         )
 
+    def transform_table(self, table_name: str, in_df: pd.DataFrame) -> pd.DataFrame:
+        """FERC Form 2 specific table transformations.
+
+        Remove duplicate IDs from the table enumerating all respondents, retaining the
+        most recently reported version of the record. Assumes that records have been
+        added to the DB in chronological order.
```

Comment:
Is this a valid assumption?  Would it be more robust and also easy for us to inject a `report_yr` column into this table as it's built up, sort by `report_yr` and `respondent_id`, drop duplicates, and then drop the `report_yr` column?

Reply:
I think I've implemented something like what you're suggesting here, if you'd take a look!

Miraculously it seems to have worked on the first try and not broken anything? Very suspicious.